azure/bastion/main.tf

```diff
@@ -2,26 +2,22 @@
 # Resource Group
 #################
 
-resource "azurerm_resource_group" "main" {
-  count = var.resource_group_name == null ? 1 : 0
+ resource "azurerm_resource_group" "main" {
+   count = var.resource_group_name != null ? 1 : 0
```

Comment:
With this it's not really an optional resource, since everything else has `var.resource_group_name` as its resource group. Think it'll probably cause issue as there's then no dependency. Probably best just removing the resource entirely and using an external resource group.